shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -633,48 +611,33 @@ private void sendViewportMetricsToFlutter() {
 
   @Override
   public boolean onKeyUp(int keyCode, KeyEvent event) {
-    if (!isAttachedToRenderer) {
+    if (!isAttachedToFlutterEngine) {
       return super.onKeyUp(keyCode, event);
     }
 
-    Map<String, Object> message = new HashMap<>();
-    message.put("type", "keyup");
-    message.put("keymap", "android");
-    encodeKeyEvent(event, message);
-    mFlutterKeyEventChannel.send(message);
+    flutterEngine.getSystemChannels().keyEvent.keyUp(event);
     return super.onKeyUp(keyCode, event);
   }
 
   @Override
   public boolean onKeyDown(int keyCode, KeyEvent event) {
-    if (!isAttachedToRenderer) {
+    if (!isAttachedToFlutterEngine) {
       return super.onKeyDown(keyCode, event);
     }
 
     if (event.getDeviceId() != KeyCharacterMap.VIRTUAL_KEYBOARD) {
-      if (mLastInputConnection != null && mImm.isAcceptingText()) {
-        mLastInputConnection.sendKeyEvent(event);
+      if (lastInputConnection != null && imm.isAcceptingText()) {
+        lastInputConnection.sendKeyEvent(event);
       }
     }
 
-    Map<String, Object> message = new HashMap<>();
-    message.put("type", "keydown");
-    message.put("keymap", "android");
-    encodeKeyEvent(event, message);
-    mFlutterKeyEventChannel.send(message);
+    flutterEngine.getSystemChannels().keyEvent.keyDown(event);
     return super.onKeyDown(keyCode, event);
   }
-
-  private void encodeKeyEvent(KeyEvent event, Map<String, Object> message) {
-    message.put("flags", event.getFlags());
-    message.put("codePoint", event.getUnicodeChar());
-    message.put("keyCode", event.getKeyCode());
-    message.put("scanCode", event.getScanCode());
-    message.put("metaState", event.getMetaState());
-  }
   //------ END VIEW OVERRIDES ----
 
   //------ START SPLASH CONTROL -----
+  @SuppressWarnings("unused")
```

Comment:
Why is this necessary on a public method in library code? I'm not very familiar with Android linters.